src/components/AttachmentPicker/index.native.tsx

```diff
@@ -256,22 +256,25 @@ function AttachmentPicker({type, children, shouldHideCameraOption}) {
     /**
      * Handles the image/document picker result and
      * sends the selected attachment to the caller (parent component)
-     *
-     * @param {Array<ImagePickerResponse|DocumentPickerResponse>} attachments
-     * @returns {Promise}
      */
     const pickAttachment = useCallback(
-        (attachments = []) => {
+        (attachments: Array<(Asset & DocumentPickerResponse) | void> = []): Promise<void> | undefined => {
             if (attachments.length === 0) {
                 onCanceled.current();
                 return Promise.resolve();
             }
-            const fileData = _.first(attachments);
-            if (Str.isImage(fileData.fileName || fileData.name)) {
-                RNImage.getSize(fileData.fileCopyUri || fileData.uri, (width, height) => {
+
+            const fileData = attachments[0];
+
+            if (!fileData) {
+                onCanceled.current();
+                return Promise.resolve();
+            }
+            if (fileData.fileName && Str.isImage(fileData.fileName ?? fileData.name)) {
```

Comment:
This will cause regression when `fileName` doesn't exist but `name` exists.
Revert to original logic